### ISSUE TITLE
Have a way to look at small stacks

### ISSUE THREAD

**Issue by eh-am** [pyroscope-io/pyroscope#1009]
Sometimes there're stacktraces with small samples, which are not possible to "drill down" in the UI.
![image](https://user-images.githubusercontent.com/6951209/162270047-d39af1f5-1023-472f-afec-5611ae6837d1.png)

Although for most use cases we want to look at the "bigger" ones, it's still useful to drill down the smaller ones, specially in the context of exploration.

Suggestions of UI approaches are welcome.

**Comment by xtrime-ru** [comment]
We got same issue. With many independent cli scripts its very hard to work with flamegraph.

Our reglar workflow is: 

1. Find method in table and click on it to make it hightlighted in flamegraph. 
2. Open flamegraph in full width and increase browser width until selected block became visible. 

Is it possible to add some option in config to make all blocks selectable / highlightable?  Or add some button to focus on selected function on flamegraph?

**Comment by Rperry2174** [comment]
Now that we've added #1636 I think we can revisit this with another "viewing" mode that we could call something like "inspector" mode where your cursor turns into a magnifying glass and then you select the area where you would like to zoom:
![inspector_mode_01](https://user-images.githubusercontent.com/23323466/203868421-7f49d3f8-267f-4131-b7e6-8aecef6df17c.png)

Selection could look something like this:
![heatmap_vid_01](https://user-images.githubusercontent.com/23323466/203868456-79503fa8-db82-4ed8-a636-ed02a7d47db0.gif)

Then after you have selected an area the canvas will "zoom" so that that area is the only area you are looking at. Later if we want to get fancy we could make it so that when the flamegraph is in a "zoom" state that you can navigate it using a hand cursor and grabbing and moving the flamegraph around

(similar to adobe suite)
![Screen Recording 2022-11-24 at 5 04 10 PM](https://user-images.githubusercontent.com/23323466/203868687-f0f8c869-e944-4c9d-9e5f-f3ea07885e48.gif)




wdyt @eh-am ? @xtrime-ru ?

**Comment by xtrime-ru** [comment]
@Rperry2174 Looks awesome!

Off-top question: How can we get rid of "other" in flamegrapgh?
See it quite often in deep traces: 
<img width="656" alt="Снимок экрана 2022-11-24 в 23 20 26" src="https://user-images.githubusercontent.com/34161928/203869932-7c734c12-2e9a-4b7a-96ee-031829623cbe.png">

**Comment by Rperry2174** [comment]
@xtrime-ru this typically happens when the trees are too large. Normally the defaults work fine, but in some cases they might lead to this issue with excessive “other” nodes. There are two settings that affect this:

- `max-nodes-serialization`
- `max-nodes-render`

The first one is used when saving data on disk, the second one is used when rendering data. You can see which limit you’re currently hitting by adding `&maxNodes=10000000` to the URL — if you’re hitting max-nodes-render value the “other” nodes will disappear, if you’re hitting `max-nodes-serialization` they will still be there.

One other thing I should mention is that in version `0.32.0` the serialization algorithm was improved so that even with default settings we do not truncate the trees as much as we used to, preserving the tree structure for as long as possible.

In addition to that, we’ve added support for -1 value for both `max-nodes-render` and `max-nodes-serialization` parameters. This allows you to disable the truncation of the trees altogether.

We’ve done tests with an app that has many code paths and pretty diverse trees and the results were good. However, using -1 is definitely a tradeoff, so here’s a few things to keep in mind:

- Setting max-nodes-serialization to -1 will increase storage requirements. Good news is that it does not significantly increase CPU usage in ingestion path.
- Setting max-nodes-render to -1 will make rendering slower. Hard to put a number on this one as it depends on your internet connection and your computer

I would not recommend setting both parameters to -1 if you have applications that generate dynamic code paths. In this case I would just set the values higher than defaults (default for serialization is 2048 and render is 8192)
---
So to summarize, my recommendations are as follows:

- Definitely upgrade to the latest version (0.34.0). This shouldn’t affect the performance much, but it will improve the accuracy of the flamegraphs.
- Consider setting `max-nodes-serialization` and `max-nodes-render parameters` to -1 or just higher values. See the tradeoffs described above.

Hope that helps, and let me know if you have any other questions!

**Comment by xtrime-ru** [comment]
@Rperry2174 
Thanks for such detailed answer. Ive increased max-nodes-serialization and max-nodes-render in 2 times in config and already can see, "other" items share reduced in 2 times.

Also i tried new "sandwich" mode and it helps alot with finding the right path!
